lib/charms/sdcore_upf/v0/fiveg_n3.py

```diff
@@ -260,6 +260,6 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
             event (RelationChangedEvent): Juju event
         """
         relation_data = event.relation.data
-        upf_ip_address = relation_data[event.unit].get("upf_ip_address")  # type: ignore[index]
+        upf_ip_address = relation_data[event.app].get("upf_ip_address")  # type: ignore[index]
```

Comment:
shouldn't we check if unit is leader?

Reply:
I think, we do not need this check here. Because non-leaders can read from other application databags (event.relation.data[event.app]).  And those are all read actions.

Reply:
As discussed in the chat, I don't think it's needed here as this is a requirer side. I don't think we care who reads the data. We could be checking leadership on the provider side, but then:
1. Should this be handled by the charm or lib?
2. Do we need it now if we don't support scaling?